docs/_spec/04-basic-declarations-and-definitions.md

```diff
@@ -218,17 +219,21 @@ A variable definition `var ´x_1, ..., x_n: T´ = ´e´` is a shorthand for the
 ## Type Declarations and Type Aliases
 
 ```ebnf
-Dcl        ::=  ‘type’ {nl} TypeDcl
-TypeDcl    ::=  id [TypeParamClause] [‘>:’ Type] [‘<:’ Type]
-Def        ::=  ‘type’ {nl} TypeDef
-TypeDef    ::=  id [TypeParamClause] ‘=’ Type
+Dcl             ::=  ‘type’ {nl} TypeDcl
+TypeDcl         ::=  id [TypeParamClause] [‘>:’ Type] [‘<:’ Type]
+Def             ::=  ‘type’ {nl} TypeDef
+                  |  ‘opaque‘ ‘type‘ {nl} OpaqueTypeDef
+TypeDef         ::=  id [TypeParamClause] ‘=’ Type
+OpaqueTypeDef   ::=  id [TypeParamClause] [‘>:’ Type] [‘<:’ Type] ‘=’ Type
 ```
 
 A possibly parameterized _type declaration_ `type ´t´[´\mathit{tps}\,´] >: ´L´ <: ´H´` declares ´t´ to be an abstract type.
 If omitted, ´L´ and ´H´ are implied to be `Nothing` and `scala.Any`, respectively.
 
 A possibly parameterized _type alias_ `type ´t´[´\mathit{tps}\,´] = ´T´` defines ´t´ to be a concrete type member.
 
+An _opaque type alias_ `type ´t´ >: ´L´ <: ´H´ = ´T´` defines ´t´ to be an opaque type alias with public bounds `>: ´L´ <: ´H´` and a private alias `= ´T´`.
```

Comment:
there is not a mention anywhere of `opaque` being a soft modifier

Reply:
That's in #17943.

Reply:
ok